File: cloudops-agent-backend/internal/domain/cloud_service.go
package domain

import "time"

type CloudServiceStatus string

const (
	CloudServiceStatusRunning    CloudServiceStatus = "running"
	CloudServiceStatusRestarting CloudServiceStatus = "restarting"
	CloudServiceStatusDown       CloudServiceStatus = "down"
)

type CloudService struct {
	ID       string             `json:"id"`
	Name     string             `json:"name"`
	Status   CloudServiceStatus `json:"status"`
	CPUUsage float64            `json:"cpuUsage"`
	Logs     []CloudServiceLog  `json:"logs"`
}

type CloudServiceLog struct {
	DateTime time.Time `json:"dateTime"`
	Event    string    `json:"event"`
}

File: cloudops-agent-frontend/src/types/cloudService.ts
export const CloudServiceStatus = {
  Running: "running",
  Restarting: "restarting",
  Down: "down",
} as const;

export type CloudServiceStatus =
  (typeof CloudServiceStatus)[keyof typeof CloudServiceStatus];

export interface CloudServiceLog {
  dateTime: string;
  event: string;
}

export interface CloudService {
  id: string;
  name: string;
  status: CloudServiceStatus;
  cpuUsage: number;
  logs: CloudServiceLog[];
}
